src/node_i18n.cc

```diff
@@ -287,6 +287,14 @@ void Transcode(const FunctionCallbackInfo<Value>&args) {
   const enum encoding toEncoding = ParseEncoding(isolate, args[2], BUFFER);
 
   if (SupportedEncoding(fromEncoding) && SupportedEncoding(toEncoding)) {
+    // If encodings are the same then there is nothing to transcode and we can
+    // just copy the buffer
+    if (fromEncoding == toEncoding) {
+      MaybeStackBuffer<char> stackbuf(input.length());
+      memcpy(stackbuf.out(), input.data(), input.length());
+      args.GetReturnValue().Set(Buffer::New(env, &stackbuf).ToLocalChecked());
```

Comment:
There's any existing method for `Buffer::Copy(...)` isn't there? Or am I remembering wrong?

Reply:
Would it be worth adding one? It seems like such an optimization could be fairly common so we could benefit from one. I lack sufficient context to be able to tell, maybe worthwhile adding it if I see this repeat itself in another place

Reply:
You can do it in a follow-up pull-request.

Reply:
@jasnell isn't remembering incorrectly, using the existing `Buffer::Copy()` is both simpler and faster in this case

Reply:
Thank you @addaleax, I will use that!